src/core/sys/openbsd/execinfo.d

```diff
@@ -0,0 +1,133 @@
+/**
+ * NetBSD implementation of glibc's $(LINK2 http://www.gnu.org/software/libc/manual/html_node/Backtraces.html backtrace) facility.
+ *
+ * Copyright: Copyright Martin Nowak 2012.
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:   Martin Nowak
+ * Source:    $(DRUNTIMESRC core/sys/netbsd/_execinfo.d)
+ */
+module core.sys.openbsd.execinfo;
+
+version (OpenBSD):
+extern (C):
+nothrow:
+
+import core.sys.openbsd.dlfcn;
+
+// Use extern (D) so that these functions don't collide with libexecinfo.
+
+extern (D) int backtrace(void** buffer, int size)
+{
+    import core.thread : thread_stackBottom;
+
+    void** p, pend=cast(void**)thread_stackBottom();
+    version (D_InlineAsm_X86)
+        asm nothrow @trusted { mov p[EBP], EBP; }
+    else version (D_InlineAsm_X86_64)
+        asm nothrow @trusted { mov p[RBP], RBP; }
+    else
+        static assert(false, "Architecture not supported.");
```

Comment:
Assuming that this is a copy of NetBSD, it's a pity that this non portable code got accepted.

Reply:
I'll need to look at the backtrace stuff again.